route/map.js

```diff
@@ -46,24 +47,55 @@ map.on('mouseleave', 'legs', function() {
  * @param coordinates {array} Array of coordinates
  * @param legs {array} route legs (stops) - each leg represents either a charging station, or via point or final point
  */
-export const drawRoute = (coordinates, legs) => {
+export const drawRoute = routes => {
   if (map.loaded()) {
-    drawPolyline(coordinates);
-    showLegs(legs);
+    for (let i = routes.length - 1; i >= 0; i--) {
+      drawPolyline(routes, i);
```

Comment:
A function should only have access to the input arguments that it really needs. Instead of passing all of the routes and i (which is a bad name I think), it would be better to refactor that function to accept one single route. 

Instead of doing a for loop you can then simpy do `routes.forEach((route, index) => drawPolyline({route, lineColor:  index === 0 ? '#fff' : '#000'}); `